From 4fe5971b008cabb1b83e992f7aa0613fcfc6cc69 Mon Sep 17 00:00:00 2001
From: "Sean P. McDonald" <sean.mcdonald@puppetlabs.com>
Date: Fri, 10 Aug 2018 15:51:00 -0700
Subject: [PATCH] Replace reference to RUBY var with
 /opt/pl-build-tools/bin/ruby

This is the Solaris only version of this patch. We do not ship pl-ruby for
solaris-10-sparc, so this patch uses system ruby instead.

This is probably the worst hack I have managed yet: in our cross compiled
builds for some reason when ruby is called with -I$(root build dir) the build
fails because the ruby we are using to build loads the rbconfig for the ruby
we are building.

ALL of the variables that reference ruby available to the depends files in
ext/etc and ext/ripper call ruby with $(ruby location) -I$(root build dir) so
every time those files are called from make there's a failure (because, as
mentioned before, the ruby using to build loads the rbconfig for the ruby
that's being built)

I don't understand what changed between ruby 2.4.4 and ruby 2.5.1 that causes
this behavior, it could be that the ruby build is putting the rbconfig in a
new place, or it could be that there's some terrible makefile magic going on
that ruby 2.4.4 builds weren't appending -I$(root build dir) to the RUBY
variable.

Whatever the cause, this patch just the use of the fully qualified path to
pl-ruby: /opt/pl-build-tools/bin/ruby instead of referencing the RUBY var
---
 ext/etc/depend    | 2 +-
 ext/ripper/depend | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/ext/etc/depend b/ext/etc/depend
index 2d986c5913..a19183b7f7 100644
--- a/ext/etc/depend
+++ b/ext/etc/depend
@@ -1,6 +1,6 @@
 constdefs.h : $(srcdir)/mkconstants.rb
 	@echo "generating constant definitions"
-	@$(RUBY) $(srcdir)/mkconstants.rb -o constdefs.h
+	@/opt/csw/bin/ruby $(srcdir)/mkconstants.rb -o constdefs.h

 # AUTOGENERATED DEPENDENCIES START
 etc.o: $(RUBY_EXTCONF_H)
diff --git a/ext/ripper/depend b/ext/ripper/depend
index 5d3bce4ec3..0d6a3a891b 100644
--- a/ext/ripper/depend
+++ b/ext/ripper/depend
@@ -29,7 +29,7 @@ check: .eventids2-check

 .eventids2-check: $(GEN) $(SRC1) $(SRC2)
 	$(ECHO) checking $(SRC1) and $(SRC2)
-	$(Q) $(RUBY) $(GEN) --mode=check --ids1src=$(SRC1) --ids2src=$(SRC2)
+	$(Q) /opt/csw/bin/ruby $(GEN) --mode=check --ids1src=$(SRC1) --ids2src=$(SRC2)
 	@exit > $@

 eventids1.c: $(srcdir)/tools/generate.rb $(SRC1)
--
2.18.0.windows.1

